### PR TITLE
CF-534 - productize tiamat

### PR DESCRIPTION
**Pull request by usnavi**
- adding run script for tiamat

### CODE REVIEW

File: tiamat/src/main/resources/tiamat.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+SPARK_HOME=/usr/hdp/2.2.0.0-2041/spark
+TIAMAT_HOME=/opt/cloudfeeds-nabu/tiamat
+
+${SPARK_HOME}/bin/spark-submit --class com.rackspace.feeds.archives.Tiamat --master yarn-client ${TIAMAT_HOME}/lib/cloudfeeds-nabu-tiamat-*-SNAPSHOT-all.jar "$@"
```
Comment (shintasmith) [inline tiamat/src/main/resources/tiamat.sh]: Are our jar files gonna always have -SNAPSHOT in it?

Comment (usnavi) [inline tiamat/src/main/resources/tiamat.sh]: doh!  I'll fix that.

**Comment by usnavi** [comment]
@shintasmith - fixed jar reference.

File: tiamat/src/main/resources/tiamat.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+SPARK_HOME=/usr/hdp/2.2.0.0-2041/spark
+TIAMAT_HOME=/opt/cloudfeeds-nabu/tiamat
+
+${SPARK_HOME}/bin/spark-submit --class com.rackspace.feeds.archives.Tiamat --master yarn-client ${TIAMAT_HOME}/lib/cloudfeeds-nabu-tiamat-*.jar "$@"
```
Comment (shintasmith) [inline tiamat/src/main/resources/tiamat.sh]: I have another question. Are we always making sure that there's only one jar file in the directory mentioned? What is the RPM going to install? Is it going to install a file cloudfeeds-nabu-tiamat-$version.jar ? If so, then as newer RPMs are installed, are you removing the old version? I don't think RPM will automatically do that. You will have to put some postinstall or preinstall script.

The way we do it in Preferences Svc is that when the RPM is installed, we always create a soft link of preferences-svc.jar to the actual preferences-svc-$version.jar that's installed with the RPM. And throughout the script, we only use/reference the preferences-svc.jar (without version). You can take a look at the gradle script there.

Usmu didn't have this problems cuz we are not using version in filenames :-P

Comment (ChandraAddala) [inline tiamat/src/main/resources/tiamat.sh]: "cloudfeeds-nabu-tiamat-*.jar" seems like a risky thing to do.

**Comment by usnavi** [comment]
@shintasmith  - added jar link for tiamat.sh, fixed exclude dependencies for build script.  Should be ready to go.

**Comment by shintasmith** [comment]
lgtm;

File: tiamat/build.gradle [hunk 3]
```diff
@@ -112,6 +111,22 @@ ospackage {
...
+        fileMode 0755
+    }
+
+    // check to see which jar got built for our symlink so the init script works
+    def version_jar = packageName + '-' + version_clean + '-all.jar'
+    // println version_jar
+
+    // create a symlink to named <app_name>.jar to the versioned jar
+    // FIXME: It should be noted that this won't work with -SNAPSHOT versions
+    link( install_path + '/lib/' + packageName + '.jar', install_path + '/lib/' + version_jar)
```
Comment (ChandraAddala) [inline tiamat/build.gradle]: the FIXME: comment above link statement. I was able to take care of that in ballista by doing this.

buildRpm.doFirst {
    if (project.version.contains('SNAPSHOT')) {
        release = 'SNAPSHOT'
        version = project.version.minus("-SNAPSHOT")
    } else {
        release = 1
        version = project.version
    }

    def version_jar = rpmProps.package_name + '-' + project.version + '.jar'
    
    // create a symlink to named <app_name>.jar to the versioned jar
    link(rpmProps.prefix_path + '/lib/' + rpmProps.app_name + '.jar', rpmProps.prefix_path + '/lib/' + version_jar)
}

Comment (usnavi) [inline tiamat/build.gradle]: cool!  I'll update that.

**Comment by usnavi** [comment]
@ChandraAddala - updated the install to handle -SNAPSHOT deployments
@shintasmith - deleted the util directory with the dev-test code you hated so much. :-P

**Comment by shintasmith** [comment]
lgtm, I will let @ChandraAddala review and merge

**Pull request merged**